### ISSUE TITLE
Bug/Debian package half installed if port `8080` is already in use ?

### ISSUE THREAD

**Issue by buixor** [crowdsecurity/crowdsec#771]
As per reported on twitter : https://twitter.com/tech_jools/status/1388084037340868612

If the port `8080` is already in use, package install fails (it's logic, although we want to change the default port in next major release ... it's a breaking change).
The bug is that apparently it lets the package in a half broken state ...

**Comment by buixor** [comment]
we're going to mimic what others (ie. nginx) are doing :
 - in postinst, we check that the mandatory ports (`8080`) is in use, and if so, we don't try to start the service

**Comment by buixor** [comment]
this is how nginx does it :

```bash
if [ -x /etc/init.d/nginx ]; then
  if [ -f /run/nginx.pid ] && pidof /usr/sbin/nginx >/dev/null; then
    invoke-rc.d nginx upgrade || invoke-rc.d nginx restart
    exit $?
  else
    if [ -z "$(ss -nlt 'sport = 80' | grep -v ^State)" ]; then
      invoke-rc.d nginx start || exit $?
    else
      echo "Not attempting to start NGINX, port 80 is already in use."
      exit 0
    fi
  fi
fi
```

**Comment by registergoofy** [comment]
Fixed in the next package release.

**Comment by sabban** [comment]
Fixed in the next packaging release.

**Comment by buixor** [comment]
available in the 1.1.0 release with debian packages
